cmdx.py

```diff
@@ -2627,7 +2674,27 @@ def __str__(self):
         return str(self.read())
 
     def __repr__(self):
-        return str(self.read())
+        try:
+            # Delegate the value reading to __str__
+            read_result = str(self)
```

Comment:
Hmmmm, now str() is a call I would never expect to fail, and if it does then I'd consider that a bug. We can expect it will never fail, and not bother with a `valid = True`.

Reply:
> PS: I feel like I might've opened a can of worms, but running bs['inputTarget'] causes an API failure when it tries to get 'targetBindMatrix'. I might create a separate issue on it so we keep on topic here.

This is the reason why I added a try/except. running the following causes an API failure:

```python
>>> _ = cmds.file(new=1, f=1)
>>> sphere_a = cmds.polySphere(ch=False)[0]
>>> sphere_b = cmds.polySphere(ch=False)[0]
>>> bs = cmds.blendShape(sphere_a, sphere_b)[0]
>>> bs = cmdx.ls(bs)[0]
>>> bs["inputTarget"].read()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\repos\cmdx\cmdx.py", line 4044, in read

  File "D:\repos\cmdx\cmdx.py", line 5076, in _plug_to_python
    #
  File "D:\repos\cmdx\cmdx.py", line 5094, in _plug_to_python
    )
  File "D:\repos\cmdx\cmdx.py", line 5094, in <genexpr>
    )
  File "D:\repos\cmdx\cmdx.py", line 5076, in _plug_to_python
    #
  File "D:\repos\cmdx\cmdx.py", line 5094, in _plug_to_python
    )
  File "D:\repos\cmdx\cmdx.py", line 5094, in <genexpr>
    )
  File "D:\repos\cmdx\cmdx.py", line 5119, in _plug_to_python
    # E.g. transform["worldMatrix"][0]
RuntimeError: (kFailure): Unexpected Internal Failure
```

Might be an edge case, but that was the reason why I added tat in.

I did some digging and it _could_ be because the attribute which it fails on (`blendshape1.inputTarget[0].inputTargetGroup[0].targetBindMatrix`) has its value being cached. It is just a theory at this point though, I haven't properly diagnosed this.
```python
# Isolated API fail:
>>> bs['inputTarget'][0][0][0][4]._mplug.asMObject()             
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: (kFailure): Unexpected Internal Failure

>>> bs['inputTarget'][0][0][0][4]._mplug.isCaching
True
```

We could remove the try/except and deal with this as a separate issue though, which is probably a cleaner strategy.